js/src/wrappers/oai.ts

```diff
@@ -145,7 +147,43 @@ interface NonStreamingChatResponse {
     | undefined;
 }
 
-function wrapBetaChatCompletion<
+function wrapBetaChatCompletionParse<
+  P extends ChatParams,
+  C extends Promise<NonStreamingChatResponse>,
+>(completion: (params: P) => C): (params: P) => Promise<any> {
+  return async (allParams: P & SpanInfo) => {
+    const { span_info: _, ...params } = allParams;
+    const span = startSpan(
+      mergeDicts(
+        {
+          name: "Chat Completion",
+          spanAttributes: {
+            type: SpanTypeAttribute.LLM,
+          },
+        },
+        parseChatCompletionParams(allParams),
+      ),
+    );
+    const startTime = getCurrentUnixTimestamp();
+    const ret = await completion(params as P);
+    try {
+      span.log({
+        output: ret.choices,
+        metrics: {
+          time_to_first_token: getCurrentUnixTimestamp() - startTime,
+          tokens: ret.usage?.total_tokens,
+          prompt_tokens: ret.usage?.prompt_tokens,
+          completion_tokens: ret.usage?.completion_tokens,
+        },
+      });
```

Comment:
should we refactor this into a common function since it's shared with the normal chat completion stuff?